File: atcoder/abc159_d.py
# https://atcoder.jp/contests/abc159/tasks/abc159_d
import sys
# input = sys.stdin.buffer.readline
def input(): return sys.stdin.readline().rstrip()
sys.setrecursionlimit(10 ** 7)

from collections import Counter
N = int(input())
A = list(map(int, (input().split())))
c = Counter(A)
def nC2(n):
    return n*(n-1)//2

all_cnt = 0
for k, v in c.items():
    if v>1:
        all_cnt += nC2(v)

for i in range(N):
    k = c[A[i]]
    ans = all_cnt - nC2(k) + nC2(k-1)
    print(ans)

# S = input()

# N, M = map(int, input().split())
# ans = (N+M)*(N+M-1)//2 - N*M
# print(ans)

# A = [[int(i) for i in input().split()] for _ in range(N)]

File: atcoder/abc159_d.cpp
// https://atcoder.jp/contests/abc159/tasks/abc159_d
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define drep(i,n) for(int i = (n-1); i >= 0; i--)
#define all(v) (v).begin(),(v).end()
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

int main() {
    int MAXN = 200000;
    int n;
	cin >> n;
    vector<int> a(n);
    rep(i,n) cin >> a[i];
    vector<ll> cnt(MAXN);
    rep(i,n) cnt[a[i]-1]++;
    ll ans = 0;
    rep(i,MAXN) {
        if (cnt[i] >= 2) {
            ans += (ll)cnt[i]*(cnt[i]-1)/2;
        }
    }
    rep(i,n) {
        cout << ans - (cnt[a[i]-1]-1) << endl;
    }
	return 0;
}
